x-pack/plugins/observability/public/config/register_alerts_table_configuration.tsx

```diff
@@ -36,7 +40,7 @@ const getO11yAlertsTableConfiguration = (
     },
   ],
   useActionsColumn: getRowActions(observabilityRuleTypeRegistry, config),
-  useBulkActions: useBulkAddToCaseActions,
+  useBulkActions: useBulkAddToCaseActions as UseBulkActionsRegistry,
```

Comment:
This is going to be a problem, I am looking at the new type who looks something like that
```ts 
export interface BulkActionsConfig {
  label: string;
  key: string;
  'data-test-subj'?: string;
  disableOnQuery: boolean;
  disabledLabel?: string;
  onClick: (
    selectedIds: TimelineItem[],
    isAllSelected: boolean,
    setIsBulkActionsLoading: (isLoading: boolean) => void,
    clearSelection: () => void,
    refresh: () => void
  ) => void;
}

export type UseBulkActionsRegistry = (
  query: Pick<QueryDslQueryContainer, 'bool' | 'ids'>
) => BulkActionsConfig[];

```
and now in o11y in this file `x-pack/plugins/observability/public/hooks/use_alert_bulk_case_actions.ts` their actions are define like that, I am pretty sure it won't work!
```ts
onClick: (items?: TimelineItem[]) => {
    const caseAttachments = items ? casesUi.helpers.groupAlertsByRule(items) : [];
    createCaseFlyout.open({ attachments: caseAttachments });
},
```

Reply:
using `cast` it is like putting the dust under the carpet ;), we need to be careful.

Reply:
@XavierM , you are right but the problem is that...these actions( eg, useAddToCase actions), specially in security solution work at multiple places ( dataTable and trigger actions table)  with slightly different formats. But both of them work because, they get `items` as a common parameter in their actions.

The other solution will be to make 2 copies of the same function one conforming to Trigger Actions and other previous version.

Now in 011y, I did  not want to modify their original implementation but with this casting it will stop question about the type. It still works as I have tested it.

I can go ahead and modify o11y's implementation to match trigger actions but I am not sure of the impact.

Reply:
@XavierM , created a `type-safe` wrapper for trigger actions over existing action hook in `o11y`.

Reply:
@XavierM , it is resolved now. please check the latest update. I have created the wrapper around `o11y` hook which is type-safe.

Reply:
thank you!